cmd/crank/beta/validate/manager_test.go

```diff
@@ -0,0 +1,176 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validate
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/google/go-containerregistry/pkg/v1/static"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	conregv1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/spf13/afero"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+)
+
+var (
+	configDep2Yaml = []byte(`apiVersion: meta.pkg.crossplane.io/v1alpha1
+kind: Configuration
+metadata:
+  name: config-dep-2
+spec:
+  dependsOn:
+    - provider: provider-dep-1
+      version: "v1.3.0"
+    - function: function-dep-1
+      version: "v1.3.0"
+---
+
+`)
+
+	providerYaml = []byte(`apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-dep-1
+spec:
+  package: provider-dep-1:v1.3.0
+---
+
+`)
+
+	funcYaml = []byte(`apiVersion: pkg.crossplane.io/v1beta1
+kind: Function
+metadata:
+  name: function-dep-1
+spec:
+  package: function-dep-1:v1.3.0
+---
+
+`)
+)
+
+func TestConfigurationTypeSupport(t *testing.T) {
+	fs := afero.NewMemMapFs()
+	w := &bytes.Buffer{}
+
+	m := NewManager(".crossplane/cache", fs, w)
+	confpkg := static.NewLayer(configDep2Yaml, types.OCILayer)
+	pd := static.NewLayer(providerYaml, types.OCILayer)
+	fd := static.NewLayer(funcYaml, types.OCILayer)
+
+	type args struct {
+		extensions []*unstructured.Unstructured
+		fetchMock  func(image string) (*conregv1.Layer, error)
+	}
+	type want struct {
+		err   error
+		confs int
+		deps  int
+	}
+	cases := map[string]struct {
+		reason string
+		args   args
+		want   want
+	}{
+		"SuccessfulDependenciesAddition": {
+			reason: "All dependencies should be successfully added from both Configuration.pkg and Configuration.meta",
+			args: args{
+				extensions: []*unstructured.Unstructured{
+					{
+						Object: map[string]interface{}{
+							"apiVersion": "meta.pkg.crossplane.io/v1alpha1",
+							"kind":       "Configuration",
+							"metadata": map[string]interface{}{
+								"name": "config-meta",
+							},
+							"spec": map[string]interface{}{
+								"dependsOn": []map[string]interface{}{
+									{
+										"provider": "provider-dep-1",
+										"version":  "v1.3.0",
+									},
+								},
+							},
+						},
+					},
+					{
+						Object: map[string]interface{}{
+							"apiVersion": "pkg.crossplane.io/v1alpha1",
+							"kind":       "Configuration",
+							"metadata": map[string]interface{}{
+								"name": "config-pkg",
+							},
+							"spec": map[string]interface{}{
+								"package": "config-dep-2:v1.3.0",
```

Comment:
my mental model of the dependency graph is already confused by this point 😂  - would you be able to add a comment somewhere that shows a quick little ascii tree of the dependency graph used in this test?

i tried to construct it myself, but that actually confused me a bit, because i'm not sure if there are 2 separate config packages in play here. Are the 2 configuration entries in this `extensions` list referring to the same underlying configuration (e.g. one is the pkg and one is the meta, but for the same configuration), or to two entirely separate configurations?

Reply:
it may make sense to have a 1st test case for the config `pkg` and 2nd test case for the config `meta`, so you are testing in isolation that they both work independently.

Right now, with them possibly intertwined (but i'm not sure exactly if that's the case 😇), it may be hard to tell that they are both working OK or you just happened to get all the final dependencies correct because the dependencies overlap between the two. If that was the case, one could work while the other one doesn't, yet the final result is still the same due to the overlapping dependencies.

Then in a 3rd separate case, if these config meta and config pkg are indeed separate things, it would be useful to make sure its all working when both are specified.

Reply:
I updated the tests with 3 cases which uses Configuration.meta and Configuration.pkg seperately and 3rd case with both.